src/test/kotlin/org/rust/cargo/toolchain/CargoProjectGenerationTest.kt

```diff
@@ -0,0 +1,83 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.cargo.toolchain
+
+import com.intellij.execution.process.ProcessAdapter
+import com.intellij.testFramework.fixtures.impl.TempDirTestFixtureImpl
+import org.rust.cargo.RsWithToolchainTestBase
+import org.rust.cargo.project.settings.toolchain
+import org.rust.cargo.toolchain.tools.Cargo
+import org.rust.cargo.toolchain.tools.cargo
+import org.rust.ide.newProject.RsCustomTemplate
+import org.rust.ide.newProject.RsGenericTemplate
+import org.rust.ide.newProject.RsProjectTemplate
+import org.rust.ide.newProject.makeProject
+import org.rust.openapiext.modules
+
+class CargoProjectGenerationTest : RsWithToolchainTestBase() {
+    private val tempDirFixture: TempDirTestFixtureImpl = TempDirTestFixtureImpl()
+
+    private val cargo: Cargo
+        get() = project.toolchain?.cargo()!!
+
+    override fun setUp() {
+        super.setUp()
+        tempDirFixture.setUp()
+    }
+
+    override fun tearDown() {
+        tempDirFixture.tearDown()
+        super.tearDown()
+    }
+
+    fun `test binary project generation`() = checkFilesGenerated(
+        RsGenericTemplate.CargoBinaryTemplate,
+        "Cargo.toml", "src/main.rs"
+    )
+
+    fun `test library project generation`() = checkFilesGenerated(
+        RsGenericTemplate.CargoLibraryTemplate,
+        "Cargo.toml", "src/lib.rs"
+    )
+
+    fun `test proc macro project generation`() = checkFilesGenerated(
+        RsCustomTemplate.ProcMacroTemplate,
+        "Cargo.toml", "src/lib.rs"
+    )
+
+    fun `test wasm-pack project generation`() = checkFilesGenerated(
+        RsCustomTemplate.WasmPackTemplate,
+        "Cargo.toml", "src/lib.rs", "src/utils.rs", "tests/web.rs"
+    )
+
+    fun `test cargo init with git`() {
+        cargo.init(project, testRootDisposable, tempDirFixture.getFile("")!!, "foo", false, "git")
+        assertTrue(listOf(".gitignore", ".git").map { tempDirFixture.getFile(it) }.all { it != null })
+    }
+
+    fun `test cargo init without git`() {
+        cargo.init(project, testRootDisposable, tempDirFixture.getFile("")!!, "foo", false, "none")
+        assertTrue(listOf(".gitignore", ".git").map { tempDirFixture.getFile(it) }.all { it == null })
+    }
+
+    private fun checkFilesGenerated(template: RsProjectTemplate, vararg paths: String) {
+        assertTrue(paths.map { tempDirFixture.getFile(it) }.all { it == null })
+
+        if (template is RsCustomTemplate && cargo.checkNeedInstallCargoGenerate()) {
+            cargo.installCargoGenerate(testRootDisposable, object : ProcessAdapter() {})
```

Comment:
This can slow down tests due to a long time of downloading and building of cargo-generate. I guess we can cache it on CI similarly to Evcxr

Reply:
Yep. Let's add the corresponding step to `check` workflow

Reply:
Btw `org/rustSlowTests/console/RsEvcxrTest.kt:20` just skips the tests if evcxr is not installed. Is it better to do the same here?

Reply:
Yeah, it would be great not to force anyone to install cargo generate for local test running